app/Utils/SubmissionUtils.php

```diff
@@ -30,56 +34,111 @@ public static function get_xml_type(mixed $filehandle): array
                 $file = 'CoverageJUnit';
             } elseif (str_contains($content, '<Notes')) {
                 $handler = \NoteHandler::class;
                 $file = 'Notes';
+                $schema_file = "{$schemas_dir}/{$file}.xsd";
             } elseif (str_contains($content, '<DynamicAnalysis')) {
                 $handler = \DynamicAnalysisHandler::class;
                 $file = 'DynamicAnalysis';
+                $schema_file = "{$schemas_dir}/{$file}.xsd";
             } elseif (str_contains($content, '<Project')) {
                 $handler = \ProjectHandler::class;
                 $file = 'Project';
+                $schema_file = "{$schemas_dir}/{$file}.xsd";
             } elseif (str_contains($content, '<Upload')) {
                 $handler = \UploadHandler::class;
                 $file = 'Upload';
+                $schema_file = "{$schemas_dir}/{$file}.xsd";
             } elseif (str_contains($content, '<testsuite')) {
                 $handler = \TestingJUnitHandler::class;
                 $file = 'TestJUnit';
             } elseif (str_contains($content, '<Done')) {
                 $handler = \DoneHandler::class;
                 $file = 'Done';
+                $schema_file = "{$schemas_dir}/{$file}.xsd";
             }
         }
 
         // restore the file descriptor to beginning of file
         rewind($filehandle);
 
+        // perform minimal error checking as a sanity check
+        if ($file === '') {
+            throw new CDashXMLValidationException(["ERROR: Could not determine submission"
+                                                  ." file type for: '{$xml_file}'"]);
+        }
+        if (isset($schema_file) && !file_exists($schema_file)) {
+            throw new CDashXMLValidationException(["ERROR: Could not find schema file '{$schema_file}'"
+                                                  ." to validate input file: '{$xml_file}'"]);
+        }
+
         return [
             'file_handle' => $filehandle,
             'xml_handler' => $handler,
             'xml_type' => $file,
+            'xml_schema' => $schema_file,
         ];
     }
 
+    /**
+     * Validate the given XML file based on its type
+     * @throws CDashXMLValidationException
+     */
+    public static function validate_xml(string $xml_file, string $schema_file): void
+    {
+        $errors = [];
+
+        // let us control the failures so we can continue
+        // parsing files instead of crashing midway
+        libxml_use_internal_errors(true);
+
+        // load the input file to be validated
+        $xml = new DOMDocument();
+        $xml->load($xml_file, LIBXML_PARSEHUGE);
+
+        // run the validator and collect errors if there are any
+        if (!$xml->schemaValidate($schema_file)) {
+            $validation_errors = libxml_get_errors();
+            foreach ($validation_errors as $error) {
+                if ($error->level === LIBXML_ERR_ERROR || $error->level === LIBXML_ERR_FATAL) {
```

Comment:
This line is what's causing the local testing of mine to fail.  If I remove the `LIBXML_ERR_ERROR` part of the check, the submissions are successful.  

My submission is from CTest `3.22.1` while the testing is `3.25.1`.  Otherwise the configuration files seem to be the same structure.  If I submit via CTest, it fails, but copying the content over to the Docker machine and running the manual validation passes.

@williamjallen, I'm happy to demonstrate if you'd like to see my setup

Reply:
I'm unable to replicate the issue above on further testing.  I'm not quite sure what happened 🤷